### PR TITLE
Define parameters in clean-traffic nwfilter

### PR DESCRIPTION
**Pull request by yalzhang**
Check the parameters defined in clean-traffic should take precedence. If
no parameters is defined, the mac and ip of the own interface will then
be used in the rules.

Signed-off-by: yalzhang <yalzhang@redhat.com>

### CODE REVIEW

**Comment by yalzhang** [comment]
new add 2 cases:
nwfilter_vm_start.possitive_test.default_filter.clean_traffic_with_params
nwfilter_vm_start.possitive_test.default_filter.clean_traffic_without_params

run all the cases locally and pass
$ sudo avocado run --vt-type libvirt nwfilter_vm_start
/root/code/avocado/avocado/plugins/run.py:294: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : 34c6457ebf56df23674d9c411721ed4f153863c4
JOB LOG    : /root/avocado/job-results/job-2020-08-12T22.36-34c6457/job.log
 (01/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.new_filter.variable_notation: PASS (32.33 s)
 (02/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.new_filter.ip_learning.default: PASS (41.46 s)
 (03/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.new_filter.ip_learning.kill_libvirtd: PASS (47.29 s)
 (04/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.new_filter.chain_priority_greater_than_rule: PASS (32.54 s)
 (05/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.new_filter.dhcpserver_variable: PASS (32.78 s)
 (06/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.new_filter.disallow_ipv6: PASS (34.08 s)
 (07/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.new_filter.disallow_dhcp: PASS (31.54 s)
 (08/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.new_filter.disallow_arp: PASS (33.85 s)
 (09/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.new_filter.clean_traffic: PASS (33.98 s)
 (10/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.new_filter.allow_ipv6: PASS (34.83 s)
 (11/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.new_filter.noexec_tmp_dir: PASS (20.66 s)
 (12/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.new_filter.ipset_test: PASS (33.71 s)
 (13/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.default_filter.no_other_l2_traffic: PASS (32.89 s)
 (14/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.default_filter.no_mac_spoofing: PASS (33.65 s)
 (15/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.default_filter.no_mac_broadcast: PASS (32.44 s)
 (16/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.default_filter.no_ip_spoofing: PASS (33.72 s)
 (17/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.default_filter.no_arp_spoofing: PASS (31.91 s)
 (18/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.default_filter.allow_dhcp: PASS (34.51 s)
 (19/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.default_filter.allow_ipv4: PASS (33.42 s)
 (20/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.default_filter.allow_incoming_ipv4: PASS (32.18 s)
 (21/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.default_filter.allow_arp: PASS (33.34 s)
 (22/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.default_filter.clean_traffic_with_params: PASS (33.85 s)
 (23/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.possitive_test.default_filter.clean_traffic_without_params: PASS (33.54 s)
 (24/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.negative_test.ip_learning_err: PASS (4.58 s)
 (25/25) type_specific.io-github-autotest-libvirt.nwfilter_vm_start.negative_test.dhcpserver_variable: PASS (4.59 s)
RESULTS    : PASS 25 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 785.88 s

**Review by chloerh — COMMENTED** [review]
Let's merge check_cmd_x into one list?

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py [hunk 1]
```diff
@@ -56,13 +54,51 @@ def run(test, params, env):
     filterref_dict = {}
     filterref_dict['name'] = filter_name
     filterref_dict['parameters'] = filter_param_list
+    params['filter_uuid'] = process.run("uuidgen", ignore_status=True, shell=True).stdout_text.strip()
 
+    cmd_param_list = []
```
Comment (chunfuwen) [inline libvirt/tests/src/nwfilter/nwfilter_vm_start.py:59]: Better add simple comment to explain what those codes achieve?

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py [hunk 2]
```diff
@@ -56,13 +54,51 @@ def run(test, params, env):
...
+        res = params['expect_match_%s' % seq]
+        check_dict = dict.fromkeys((cmd,), res)
+        cmd_result_list.append(check_dict)
     # backup vm xml
     vmxml_backup = libvirt_xml.VMXML.new_from_inactive_dumpxml(vm_name)
 
     libvirtd = utils_libvirtd.Libvirtd()
     device_name = None
 
+    def check_nwfilter_rules(check_cmd, expect_match):
```
Comment (chunfuwen) [inline libvirt/tests/src/nwfilter/nwfilter_vm_start.py:77]: miss doc string

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py (hunk 1 above)
Comment (yalzhang) [reply libvirt/tests/src/nwfilter/nwfilter_vm_start.py]: updated, Thank you!

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py (hunk 2 above)
Comment (yalzhang) [reply libvirt/tests/src/nwfilter/nwfilter_vm_start.py:73]: add a simple text to explain the purpose of this function

**Comment by yalzhang** [comment]
updated and rerun all the cases locally, no issue found.

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py (hunk 2 above)
Comment (chunfuwen) [reply libvirt/tests/src/nwfilter/nwfilter_vm_start.py:73]: other than simple explanation, please add doc string as review guidelines

**Review by chloerh — APPROVED** [review]

**Comment by chloerh** [comment]
Conflicted

**Comment by yalzhang** [comment]
have resolved the conflict and rerun all the cases related, all pass.

File: libvirt/tests/src/nwfilter/nwfilter_vm_start.py (hunk 2 above)
Comment (yalzhang) [reply libvirt/tests/src/nwfilter/nwfilter_vm_start.py:86]: updated, Thank you!

**Comment by yalzhang** [comment]
> Let's merge check_cmd_x into one list?

Updated, Thank you for the suggestion as I find it make the script more elegant and efficient!

**Review by chunfuwen — APPROVED** [review]
LGTM

**Pull request merged**